src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -78,14 +78,50 @@ namespace winrt::TerminalApp::implementation
     void TerminalPage::_HandleNextTab(const IInspectable& /*sender*/,
                                       const TerminalApp::ActionEventArgs& args)
     {
-        _SelectNextTab(true);
+        if (_settings->GlobalSettings().UseTabSwitcher())
+        {
+            if (CommandPalette().Visibility() == Visibility::Visible)
+            {
+                CommandPalette().SelectNextItem(true);
+            }
+
+            auto opt = _GetFocusedTabIndex();
+            uint32_t startIdx = opt ? *opt : 0;
+            startIdx = (startIdx + _tabs.Size() + 1) % _tabs.Size();
+
+            CommandPalette().EnableTabSwitcherMode(false, startIdx);
+            CommandPalette().Visibility(Visibility::Visible);
+        }
+        else
+        {
+            _SelectNextTab(true);
+        }
+
         args.Handled(true);
     }
 
     void TerminalPage::_HandlePrevTab(const IInspectable& /*sender*/,
                                       const TerminalApp::ActionEventArgs& args)
     {
-        _SelectNextTab(false);
+        if (_settings->GlobalSettings().UseTabSwitcher())
+        {
+            if (CommandPalette().Visibility() == Visibility::Visible)
+            {
+                CommandPalette().SelectNextItem(false);
+            }
+
+            auto opt = _GetFocusedTabIndex();
+            uint32_t startIdx = opt ? *opt : 0;
+            startIdx = (startIdx + _tabs.Size() - 1) % _tabs.Size();
+
+            CommandPalette().EnableTabSwitcherMode(false, startIdx);
+            CommandPalette().Visibility(Visibility::Visible);
+        }
+        else
+        {
+            _SelectNextTab(false);
+        }
+

```

Comment:
nit: looks like the only difference here is whether we pass true or false to `CommandPalette().SelectNextItem(<bool>)` and `_SelectNextTab(<bool>)`. You could probably create a helper function and refactor this a bit (along with `_SelectNextTab()`).

Reply:
... wouldn't that helper just _literally be_ `_SelectNextTab`?

Reply:
hrmmm yeah I could probably move some of this into `_SelectNextTab`